test/e2e/resize_nodes.go

```diff
@@ -287,35 +295,55 @@ func verifyPods(c *client.Client, ns, name string, wantName bool, replicas int)
-// This function executes commands on a node so it will work only for some
-// environments.
-func performTemporaryNetworkFailure(c *client.Client, ns, rcName string, replicas int, podNameToDisappear string, node *api.Node) {
-	Logf("Getting external IP address for %s", node.Name)
-	host := ""
-	for _, a := range node.Status.Addresses {
-		if a.Type == api.NodeExternalIP {
-			host = a.Address + ":22"
-			break
-		}
+func blockNetwork(from string, to string) {
+	Logf("block network traffic from %s to %s", from, to)
+	iptablesRule := fmt.Sprintf("OUTPUT --destination %s --jump REJECT", to)
+	dropCmd := fmt.Sprintf("sudo iptables --insert %s", iptablesRule)
+	if result, err := SSH(dropCmd, from, testContext.Provider); result.Code != 0 || err != nil {
+		LogSSHResult(result)
+		Failf("Unexpected error: %v", err)
 	}
-	if host == "" {
-		Failf("Couldn't get the external IP of host %s with addresses %v", node.Name, node.Status.Addresses)
+}
+
+func unblockNetwork(from string, to string) {
+	Logf("Unblock network traffic from %s to %s", from, to)
+	iptablesRule := fmt.Sprintf("OUTPUT --destination %s --jump REJECT", to)
+	undropCmd := fmt.Sprintf("sudo iptables --delete %s", iptablesRule)
+	// Undrop command may fail if the rule has never been created.
+	// In such case we just lose 30 seconds, but the cluster is healthy.
+	// But if the rule had been created and removing it failed, the node is broken and
+	// not coming back. Subsequent tests will run or fewer nodes (some of the tests
+	// may fail). Manual intervention is required in such case (recreating the
+	// cluster solves the problem too).
+	err := wait.Poll(time.Millisecond*100, time.Second*30, func() (bool, error) {
+		result, err := SSH(undropCmd, from, testContext.Provider)
+		if result.Code == 0 && err == nil {
+			return true, nil
+		}
+		LogSSHResult(result)
+		if err != nil {
+			Logf("Unexpected error: %v", err)
+		}
+		return false, nil
+	})
+	if err != nil {
+		Failf("Failed to remove the iptable REJECT rule. Manual intervention is "+
+			"required on host %s: remove rule %s, if exists", from, iptablesRule)
 	}
+}
 
-	By(fmt.Sprintf("block network traffic from node %s to the master", node.Name))
+func getMaster(c *client.Client) string {
 	master := ""
 	switch testContext.Provider {
 	case "gce":
-		// TODO(#10085): The use of MasterName will cause iptables to do a DNS
-		// lookup to resolve the name to an IP address, which will slow down the
-		// test and cause it to fail if DNS is absent or broken. Use the
-		// internal IP address instead (i.e. NOT the one in testContext.Host).
-		master = testContext.CloudConfig.MasterName
+		eps, err := c.Endpoints(api.NamespaceDefault).Get("kubernetes")
+		if err != nil {
+			Failf("Fail to get kubernetes endpoinds: %v", err)
+		}
+		if len(eps.Subsets) != 1 || len(eps.Subsets[0].Addresses) != 1 {
+			Failf("There are more than 1 endpoints for kubernetes service: %+v", eps)
+		}
+		master = eps.Subsets[0].Addresses[0].IP
```

Comment:
Use the IP instead of host name... This is to avoid external DNS dependancy for iptables.